ExecuteJobs.java

```diff
@@ -0,0 +1,124 @@
+import java.util.stream.*;
+import java.util.*;
+import java.io.*;
+//import java.util.Stream;
+
+public class ExecuteJobs {
+  private static final int ERR_SUCCESS = 0;
+  private static final int ERR_FAILED_TO_KILL = 1;
+  private static final int ERR_FAILED_TO_FIND = 2;
+  private static final int ERR_FAILED_TO_START = 3;
+
+  public ErrorInfo start(String sProcess) throws IOException {
+    ErrorInfo errorInfo = new ErrorInfo();
+    Process process = Runtime.getRuntime().exec(sProcess);
+
+    ErrorInfo errorStatus = getProcessStatus(process.pid());
+    if(errorStatus.getIoMessage() == null) {
+      errorInfo.setErrorCode(ERR_FAILED_TO_START);
+      errorInfo.setIoMessage("Failed to start " + sProcess);
+      return errorInfo;
+    }
+    errorInfo.setErrorCode(ERR_SUCCESS);
+    errorInfo.setIoMessage("Successfully start " + sProcess);
+    errorInfo.setPid(process.pid());
+    return errorInfo;
+  }
+
+  public ErrorInfo stop(long pid) throws IOException {
+    ErrorInfo errorInfo = new ErrorInfo();
+
+    ErrorInfo errorStatus = getProcessStatus(pid);
+    if(errorStatus.getIoMessage() == null) {
+      errorInfo.setErrorCode(ERR_FAILED_TO_FIND);
+      errorInfo.setIoMessage("Failed to find PID " + pid + ". PID was not killed.");
+      return errorInfo;
+    }
+
+    Runtime.getRuntime().exec("kill " + pid);
```

Comment:
As far as I understood, this returns a new `Process` object which you should `waitFor` to ensure that you're not evaluating the operation while the child process (`pid`) is still running - you cannot except all processes to terminate in an instant, some have complex shutdown sequences.
And the whole method looks kind of bloated so it is not obvious what it's responsibility is.

Reply:
I've added waitFor in several places in this class. I will also add comments to the method so that all it is more obvious what is happening in this method.